### ISSUE TITLE
Predict "Rating" column based on other features.

### ISSUE THREAD

**Issue by mihaighidoveanu** [mihaighidoveanu/data-science-lab#1]
Our data has over 1,000 missing values for the "Ratings" column. That's around 10% of the data. To not let this data go to waste, we can try to build a predictor for this column.

**Comment by mihaighidoveanu** [comment]
Until now I have tried : 
- LinearRegression : works terrible (0.1 r-squared error); maybe other combinations of features will prove better
- SVM (Support Vector Machine Regression) : we have 0.82 on the training set, but very low values for the test set. We can tune the hyperparameters of the model (C and alpha)

**Comment by mihaighidoveanu** [comment]
We can also try : 
- DecisionTrees (or Random Forests). 
- we can also transform this problem into an classification one. We can transform the continuous space of values (1-5) into certain subcategories of ratings (1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5). We will lose detail of this feature, but we may be able to build a classifier model easier than a regression one.

**Comment by mihaighidoveanu** [comment]
We transformed the problem in a classification one. Now rating can be *poor* (< 4) and *excellent* (>=4). We have around 57 % accuracy on the test data. 

We can try to improve this further. We should try adding or changing the features of data, and try more values for the hyperparameters of the algorithm

**Comment by mihaighidoveanu** [comment]
We have now a 69% accuracy using Random Forest. 
Another predictor for success on an application is the number of **Installs** it has. So we can try to model the problem as to predict it instead of **Rating**

**Comment by mihaighidoveanu** [comment]
We now have 74 % accuracy with VotingClassifier.
![image](https://user-images.githubusercontent.com/17084301/56854531-db533e80-6940-11e9-9064-0735f16de65a.png)

**Comment by mihaighidoveanu** [comment]
@claudiu-orosanu XGBoost may upgrade our performance even more

**Comment by mihaighidoveanu** [comment]
Up next : @HorridBro , @claudiu-orosanu 
We can try to combine our current strategies and see what gives the best performance.
Strategies are as follows:
cat_features - True / False
remove_outliers - True / False
upsampling - True / False
downsampling (if upsampling is False) - True / False
Combinations of the above should be tried.
